internal/services/cragcast/cragcast.go

```diff
@@ -0,0 +1,227 @@
+			return nil, err
+		}
+	}
+
+	for _, hourlyForecast := range hoursMap {
+		hours = append(hours, hourlyForecast)
+	}
+
+	sort.Slice(hours, func(i, j int) bool {
+		return hours[i].StartOfHour.Before(hours[j].StartOfHour)
+	})
+
+	return hours, nil
+}
+
+type QuantitativeValueType string
+
+const (
+	TEMPERATURE                QuantitativeValueType = "temperature"
+	HUMIDITY                   QuantitativeValueType = "relativeHumidity"
+	QUANTITATIVE_PRECIPITATION QuantitativeValueType = "quantitativePrecipitation"
+	PRECIPITATION_PROBABILITY  QuantitativeValueType = "probabilityOfPrecipitation"
+	SKY_COVER                  QuantitativeValueType = "skyCover"
+	SNOWFALL_AMOUNT            QuantitativeValueType = "snowfallAmount"
+	WIND_DIRECTION             QuantitativeValueType = "windDirection"
+	WIND_SPEED                 QuantitativeValueType = "windSpeed"
+)
+
+func processQuantitativeValues(valueType QuantitativeValueType, additionalProperties map[string]noaaTypes.GridpointQuantitativeValueLayer, hoursMap map[string]*HourlyForecast) error {
+	var value noaaTypes.GridpointQuantitativeValueLayer
+
+	switch valueType {
+	case TEMPERATURE:
+		value = additionalProperties[string(TEMPERATURE)]
+	case HUMIDITY:
+		value = additionalProperties[string(HUMIDITY)]
+	case QUANTITATIVE_PRECIPITATION:
+		value = additionalProperties[string(QUANTITATIVE_PRECIPITATION)]
+	case PRECIPITATION_PROBABILITY:
+		value = additionalProperties[string(PRECIPITATION_PROBABILITY)]
+	case SKY_COVER:
+		value = additionalProperties[string(SKY_COVER)]
+	case SNOWFALL_AMOUNT:
+		value = additionalProperties[string(SNOWFALL_AMOUNT)]
+	case WIND_DIRECTION:
+		value = additionalProperties[string(WIND_DIRECTION)]
+	case WIND_SPEED:
+		value = additionalProperties[string(WIND_SPEED)]
+	default:
+		return errors.New("invalid value type")
+	}
+
+	for _, valueIter := range value.Values {
+		time, err := validTimeToTime(valueIter.ValidTime)
+		if err != nil {
+			return err
+		}
+		stringifiedTime := time.String()
+		hourObject := hoursMap[stringifiedTime]
+		if hourObject == nil {
+			hoursMap[stringifiedTime] = &HourlyForecast{
+				StartOfHour: *time,
+			}
+			hourObject = hoursMap[stringifiedTime]
+		}
+
+		switch valueType {
+		case TEMPERATURE:
+			hourObject.TemperatureFahrenheit = celsiusToFahrenheit(valueIter.Value)
+		case HUMIDITY:
+			hourObject.Humidity = valueIter.Value
+		case QUANTITATIVE_PRECIPITATION:
+			hourObject.TotalPrecipMm = valueIter.Value
+		case PRECIPITATION_PROBABILITY:
+			hourObject.ChanceOfPrecip = valueIter.Value
+		case SKY_COVER:
+			hourObject.skyCoverPercent = valueIter.Value
+		case SNOWFALL_AMOUNT:
+			hourObject.snowfallMm = valueIter.Value
+		case WIND_DIRECTION:
+			hourObject.windDegrees = valueIter.Value
+		case WIND_SPEED:
+			hourObject.windKmH = valueIter.Value
+		default:
+			return errors.New("invalid value type")
+		}
+	}
+
+	return nil
+}
+
+func validTimeToTime(validTime noaaTypes.ISO8601Interval) (*time.Time, error) {
```

Comment:
`validTimeToTime` had to read the implementation to grok the intent here. Let's rename this to something more obvious and perhaps add a comment (if necessary) to support it.